csp_billing_adapter_microsoft/plugin.py

```diff
@@ -209,15 +209,14 @@ def _get_msi_token(config: Config):
 
     # Set resource id to the required value needed to to retrieve an
     # MSI Authentication Token
-    try:
-        config['api']
-        # if config has API key, it is running on a VM
+    if config.get('api') and config['api'] != 'no_data_query':
```

Comment:
Isn't this redundant and requires 2 lookups?
```
api_setup = config.get('api')
if api_setup and api_setup != 'no_data_query':
```
??

Reply:
Agree. Fixed.